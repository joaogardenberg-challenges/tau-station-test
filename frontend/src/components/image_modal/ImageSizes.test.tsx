import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import ThemeProvider from 'providers/Theme'

const useHistory = jest.fn()
const useLocation = jest.fn()
const useQuery = jest.fn()

jest.doMock('react-router-dom', () => ({
  useHistory,
  useLocation
}))

jest.doMock('hooks/useQuery', () => useQuery)

const ImageSizes = require('./ImageSizes').default

const renderWithTheme = (ui: any, options: any = {}) =>
  render(ui, { wrapper: ThemeProvider, ...options })

describe('Image Sizes', () => {
  const pathname = '/images/1'
  const queryToString = 'query'

  beforeEach(() => {
    useHistory.mockReturnValue({ replace: jest.fn() })
    useLocation.mockReturnValue({ pathname })
    useQuery.mockReturnValue({
      get: () => '400',
      set: jest.fn(),
      toString: () => queryToString
    })
  })

  it('renders successfully', () => {
    const { container } = renderWithTheme(<ImageSizes />)
    expect(container).toMatchSnapshot()
  })

  it("changes the url's width on button click", () => {
    const newWidth = '800'
    renderWithTheme(<ImageSizes />)

    const button800 = screen.getByRole('button', { name: `${newWidth}px` })
    userEvent.click(button800)

    expect(useQuery().set).toHaveBeenCalledTimes(1)
    expect(useQuery().set).toHaveBeenCalledWith('width', newWidth)
    expect(useHistory().replace).toHaveBeenCalledTimes(1)
    expect(useHistory().replace).toHaveBeenCalledWith({
      pathname,
      search: queryToString
    })
  })
})
