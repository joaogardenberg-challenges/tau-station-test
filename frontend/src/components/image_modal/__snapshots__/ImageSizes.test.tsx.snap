// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Sizes renders successfully 1`] = `
<div>
  <div
    class="sc-bdfBQB ekGQIK"
  >
    <div
      class="MuiButtonGroup-root MuiButtonGroup-outlined css-10mxj9p-MuiButtonGroup-root"
      role="group"
    >
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedSecondary css-gqx3nw-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        48
        px
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedSecondary css-1yrmfgj-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        400
        px
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedSecondary css-gqx3nw-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        800
        px
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedSecondary css-gqx3nw-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        1280
        px
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
